crates/client/sync/src/fetch/mod.rs

```diff
@@ -1,96 +1,111 @@
+    pub first_block: u64,
+    pub fetch_stream_sender: mpsc::Sender<UnverifiedFullBlock>,
+    pub once_caught_up_sender: oneshot::Sender<()>,
+    pub sync_polling_interval: Option<Duration>,
+    pub n_blocks_to_sync: Option<u64>,
+    pub stop_on_sync: bool,
+    pub sync_parallelism: usize,
+    pub warp_update: bool,
+    pub warp_update_port_rpc: u16,
+    pub warp_update_port_fgw: u16,
+}
+
 pub async fn l2_fetch_task(
     backend: Arc<MadaraBackend>,
-    first_block: u64,
-    n_blocks_to_sync: Option<u64>,
-    stop_on_sync: bool,
-    fetch_stream_sender: mpsc::Sender<UnverifiedFullBlock>,
     provider: Arc<GatewayProvider>,
-    sync_polling_interval: Option<Duration>,
-    once_caught_up_callback: oneshot::Sender<()>,
-    cancellation_token: tokio_util::sync::CancellationToken,
+    ctx: ServiceContext,
+    mut config: L2FetchConfig,
 ) -> anyhow::Result<()> {
     // First, catch up with the chain
-    let backend = &backend;
+    // let backend = &backend;
 
-    let mut next_block = first_block;
+    let L2FetchConfig { first_block, warp_update, warp_update_port_rpc, warp_update_port_fgw, .. } = config;
 
-    {
-        // Fetch blocks and updates in parallel one time before looping
-        let fetch_stream = (first_block..).take(n_blocks_to_sync.unwrap_or(u64::MAX) as _).map(|block_n| {
-            let provider = Arc::clone(&provider);
-            let cancellation_token = cancellation_token.clone();
-            async move {
-                (
-                    block_n,
-                    fetch_block_and_updates(&backend.chain_config().chain_id, block_n, &provider, &cancellation_token)
-                        .await,
-                )
-            }
-        });
+    if warp_update {
+        let client = jsonrpsee::http_client::HttpClientBuilder::default()
+            .build(format!("http://localhost:{warp_update_port_rpc}"))
+            .expect("Building client");
 
-        // Have 10 fetches in parallel at once, using futures Buffered
-        let mut fetch_stream = stream::iter(fetch_stream).buffered(10);
-        while let Some((block_n, val)) =
-            channel_wait_or_graceful_shutdown(fetch_stream.next(), &cancellation_token).await
-        {
-            match val {
-                Err(FetchError::Sequencer(SequencerError::StarknetError(StarknetError {
-                    code: StarknetErrorCode::BlockNotFound,
-                    ..
-                }))) => {
-                    tracing::info!("🥳 The sync process has caught up with the tip of the chain");
-                    break;
-                }
-                val => {
-                    if fetch_stream_sender.send(val?).await.is_err() {
-                        // join error
-                        break;
-                    }
-                }
-            }
+        if client.ping().await.is_err() {
+            tracing::error!("❗ Failed to connect to warp update sender on http://localhost:{warp_update_port_rpc}");
```

Comment:
shouldn't this be an anyhow::bail?

Reply:
I personally prefer this as it more clearly conveys the error to the node operator.

Reply:
Also, if I remember correctly, having this error will eventually cause the entire node to error, and I don't think that is very clean.

Reply:
isnt that what we want here?